spectrochempy/analysis/pls.py

```diff
@@ -0,0 +1,317 @@
+            warm_start=warm_start,
+            copy=True,
+            **kwargs,
+        )
+
+        # initialize sklearn PLS
+        self._pls = cross_decomposition.PLSRegression(
+            n_components=self.used_components,
+            scale=self.scale,
+            max_iter=self.max_iter,
+            copy=self.copy,
+            tol=self.tol,
+        )
+
+    # ----------------------------------------------------------------------------------
+    # Private methods (overloading abstract classes)
+    # ----------------------------------------------------------------------------------
+
+    def _fit(self, X, Y):
+        # this method is called by the abstract class fit.
+        # Input X and Y are np.ndarray
+        self._pls.fit(X, Y)
+        self._x_weights = self._pls.x_weights_.T
+        self._y_weights = self._pls.y_weights_.T
+        self._x_loadings = self._pls.x_loadings_.T
+        self._y_loadings = self._pls.y_loadings_.T
+        self._x_scores = self._pls.x_scores_
+        self._y_scores = self._pls.y_scores_
+        self._x_rotations = self._pls.x_rotations_.T
+        self._y_rotations = self._pls.y_rotations_.T
+
+        self._coef = self._pls.coef_.T
+        self._intercept = self._pls.intercept_
+        self._n_iter = self._pls.n_iter_
+        self._n_feature_in = self._pls.n_features_in_
+
+        # todo: check self.feature_names_in = self._pls.feature_names_in_
+
+        # for compatibility with superclass methods
+        self._n_components = self.used_components
+
+    def _fit_transform(self, X, Y=None):
+        # Learn and apply the dimension reduction on the train data.
+        #
+        return self._pls.fit_transform(X, Y)
+
+    # todo ?: def _get_feature_names_out([input_features])
+
+    # todo ? : def _get_params([deep])
+
+    def _inverse_transform(self, X_transform, Y_transform=None):
+        # Transform data back to its original space.
+        return self._pls.inverse_transform(X_transform, Y=Y_transform)
+
+    def _transform(self, X, Y=None, copy=True):
+        # Apply the dimension reduction.
+        return self._pls.transform(X, Y, copy)
+
+    def _predict(self, X, copy=True):
+        # Predict targets of given samples.
+        return self._pls.predict(X, copy)
+
+    def _score(self, X, Y, sample_weight=None):
+        # this method is called by the abstract class score.
+        # Input X, Y, sample_weights are np.ndarray
+        return self._pls.score(X, Y, sample_weight=sample_weight)
+
+    # ----------------------------------------------------------------------------------
+    # Public methods and properties specific to PLS
+    # ----------------------------------------------------------------------------------
+    _docstring.keep_params("analysis_fit.parameters", "X")
+
+    @_docstring.dedent
+    def fit(self, X, Y):
+        """
+        Fit the PLS model on X and Y.
+
+        Parameters
+        ----------
+        %(analysis_fit.parameters.X)s
+
```

Comment:
description of Y is missing:

    %(analysis_fit.parameters.X)s
    Y :  :term:`array-like` of shape (n_samples,) or (n_samples, n_targets)
          Target vectors, where n_samples is the number of samples and n_targets is the number of response variables.